### PR TITLE
FIX #379: web client resizes with window

### LINKED ISSUE #379: Web client is unable to detect screen resizing
I enabled browser console during loading, after disabling it I have permament reduction of screen space used by web version of AB.

Not sure is it fixable from within AB or it it an upstream limitation.

### PR DESCRIPTION
**Pull request by michaelkirk**
resize mov

### CODE REVIEW

File: map_model/src/map.rs [hunk 1]
```diff
@@ -115,21 +115,21 @@ impl Map {
     }
 
     pub fn corrupt_err(path: String, err: Box<dyn std::error::Error>) {
-        println!("\nError loading {}: {}\n", path, err);
+        error!("\nError loading {}: {}\n", path, err);
         if err.to_string().contains("No such file") {
-            println!(
+            error!(
```
Comment (michaelkirk) [inline map_model/src/map.rs:120]: I had an unrelated issue were the scenarios were failing to load, but I couldn't read the println output in the browser.

File: widgetry/src/backend_glow_wasm.rs [hunk 2]
```diff
@@ -39,6 +40,22 @@ pub fn setup(window_title: &str) -> (PrerenderInnards, winit::event_loop::EventL
     body.append_child(&canvas)
         .expect("Append canvas to HTML body");
 
+    let winit_window = Rc::new(winit_window);
```
Comment (dabreegster) [inline widgetry/src/backend_glow_wasm.rs:43]: The codebase's first `Rc`! Now this counts as "advanced Rust", right? :D

**Comment by dabreegster** [comment]
Works great, thanks for the fix! Maybe in some ideal world, winit could do this, but it maybe doesn't belong at that layer anyway.

**Comment by michaelkirk** [comment]
> Works great, thanks for the fix! Maybe in some ideal world, winit could do this, but it maybe doesn't belong at that layer anyway.

Yeah, I was surprised at first too that we had to do something this manual. But after starting the work and thinking about it a bit, I came to the conclusion that it probably shouldn't live in winit.

In the "wasm" world, a "winit::window" is a `<canvas>` element, whereas the `<window>` element is more akin to... I guess the "window manager" for a desktop program. 

e.g. in the same way we *could* have multiple windows for a given desktop app, we could have multiple *canvas* elements in a single browser tab.

Or maybe more common, some people might want the canvas element "winit::window" to be just one small part of a page.

So the fact that our entire "web app" is a single full-size canvas element is not the only use case for winit (though I imagine it's a not uncommon).

**Comment by alvinhochun** [comment]
rust-windowing/winit#1661 applies. It's hard to say exactly whether this should be a part of Winit or not, but nevertheless I believe we should figure out something so that users won't need to reimplement this exact function in every single project.

**Pull request merged**